R/target_market_share.R

```diff
@@ -282,105 +188,65 @@ target_market_share <- function(data,
       )
   }
 
-  reweighting_groups <- maybe_add_name_ald(
-    c("sector_ald", "region", "scenario", "scenario_source", "year"),
-    by_company
-  )
-
-  data <- reweight_technology_share(
-    data,
-    !!!rlang::syms(reweighting_groups)
-  )
-
-  data <- data %>%
-    pivot_wider2(
-      names_from = .data$scenario,
-      values_from = c(
-        .data$weighted_production_target,
-        .data$weighted_technology_share_target
-      )
-    )
-
-  data <- data %>%
-    rename(
-      weighted_production_projected = .data$weighted_production,
-      weighted_technology_share_projected = .data$weighted_technology_share,
-      sector = .data$sector_ald
-    )
+  data <- reweight_technology_share(data, by_company)
 
-  data <- data %>%
-    pivot_longer(cols = starts_with("weighted_")) %>%
-    filter(!is.na(.data$value)) %>%
-    separate_metric_from_name()
-
-  data <- data %>%
-    pivot_wider2()
+  data <- format_output_dataframe(data)
 
   corporate_economy <- calculate_ald_benchmark(ald, region_isos, by_company)
 
-  relevant_sectors <- unique(data$sector)
-
-  relevant_corporate_economy <- corporate_economy %>%
-    filter(.data$sector %in% relevant_sectors)
+  relevant_corporate_economy <- filter(
+    corporate_economy,
+    .data$sector %in% unique(data$sector)
+    )
 
   data %>%
     dplyr::bind_rows(relevant_corporate_economy) %>%
     ungroup()
 }
 
-pivot_wider2 <- function(data, ...) {
-  abort_if_has_list_colums(data)
-
-  out <- suppressWarnings(pivot_wider(data, ...))
-  unnest_list_columns(out)
-}
-
-unnest_list_columns <- function(data) {
-  if (utils::packageVersion("tidyr") < "1.1.2") {
-    suppressWarnings(unnest(data))
-  } else {
-    unnest(data, where(is.list))
-  }
-}
-
-tmsr_or_smsp <- function() {
-  dplyr::tribble(
-    ~which_metric, ~green_or_brown,
-    "tmsr_target_production", "brown",
-    "smsp_target_production", "green"
-  )
-}
+calculate_targets <- function(data){
+  target_groups <- c("sector_ald", "scenario", "region", "name_ald")
 
-separate_metric_from_name <- function(data) {
   data %>%
+    arrange(.data$year) %>%
+    group_by(!!!rlang::syms(c(target_groups, "technology", "year"))) %>%
+    mutate(technology_production = sum(.data$production)) %>%
```

Comment:
nit: production data seems likely to have `NA`s in it, or would it always be 0?

Reply:
It should not have any `NA`s in it. In fact, there is a (very strict) check at the beginning of `target_market_share()` that causes the function to fail if `production` contains any `NA`s. We elected to go strict rather than flexible.